### DISCUSSION TITLE
Single configuration file for home-manager and nixos

### DISCUSSION THREAD

**Discussion by noahfraiture** [nix-community/stylix#646] [Q&A] [answered]
Hello,
I found multiple references to using stylix in nixos and home manager at the same time but I don't think that fulfill my problem. I have multiple machines, some on nixos and some on other other with stylix. Currently I have a stylix configuration in nixos and I want to use it for a machine on ubuntu. My flake : 
```
      nixosConfigurations = {
        bitfenix = nixpkgs.lib.nixosSystem {
          specialArgs = {
            inherit inputs;
          };
          modules = [
            ./hosts/bitfenix/configuration.nix
            inputs.home-manager.nixosModules.home-manager
            inputs.stylix.nixosModules.stylix
            inputs.nix-flatpak.nixosModules.nix-flatpak
          ];
        };
      };
      homeConfigurations = {
        noah = inputs.home-manager.lib.homeManagerConfiguration {
          pkgs = nixpkgs.legacyPackages."x86_64-linux";
          modules = [
            ./hosts/hostinger/home.nix
            inputs.stylix.homeManagerModules.stylix
          ];
        };
      };

```
Is it possible to share a single `stylix.nix` configuration file, that look like this : 
```
{
  config,
  pkgs,
  lib,
  ...
}:
{

  options = {
    theme.option = lib.mkOption {
      type = lib.types.str;
      default = "dark";
    };
  };

  config = {
    stylix = {
      enable = true;

      image = pkgs.fetchurl {
        url = "https://gruvbox-wallpapers.pages.dev/wallpapers/anime/classroom.jpg";
        sha256 = "sha256-jzok9VIwySb+ek2soMMxXcoVUTZh8UGLqloM/npQtAo=";
      };

      # https://github.com/tinted-theming/schemes
      base16Scheme =
        if config.theme.option == "dark" then
          "${pkgs.base16-schemes}/share/themes/catppuccin-mocha.yaml"
        else if config.theme.option == "light" then
          "${pkgs.base16-schemes}/share/themes/catppuccin-latte.yaml"
        else
          throw "wrong theme : " + config.theme.option;

      polarity = config.theme.option;

      # https://github.com/ful1e5/Bibata_Cursor
      cursor = {
        package = pkgs.bibata-cursors;
        name =
          if config.theme.option == "dark" then
            "Bibata-Modern-Ice"
          else if config.theme.option == "light" then
            "Bibata-Modern-Classic"
          else
            throw "wrong theme : " + config.theme;
        size = 24;
      };

      fonts = {
        monospace = {
          package = pkgs.nerdfonts;
          name = "JetBrainsMono Nerd Font Propo";
        };
        sizes = {
          desktop = 16;
        };
      };
    };
  };
}

```
accross home-manager and nixos for different machine ?

**Comment by trueNAHO** [comment]
> Is it possible to share a single `stylix.nix` configuration file [...] accross home-manager and nixos for different machine ?

Sadly, I have never had the time to setup Stylix in NixOS and only have it setup in Home Manager. @danth, might be more helpful here. However, I assume this should already be possible in the user level with some refactoring.

**Comment by danth** [comment]
Yes, it's possible to import the same module into two independent NixOS and Home Manager instances, as long as it only mentions options which are common to both.

I'm currently doing this in my own configuration.

**Reply by noahfraiture** [reply]
Great ! Would you mind share the section where you do that ?

**Reply by danth** [reply, marked as answer]
It's just a `stylix.nix` similar to the one you showed, and then `imports = [ ./stylix.nix ];` in both NixOS and Home Manager.

I also have a few configurations [on Gist](https://gist.github.com/danth) which you can take a look at.
